src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const arrayWithObjects = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        cloneState = {
+          ...cloneState,
+          ...action.extraData,
+        };
+
+        break;
+
+      case 'removeProperties':
+        cloneState = removeProperties(cloneState, action.keysToRemove);
```

Comment:
[CODE STYLE] - don't mutate object or arrays - it will cause unexpected results later on. In the `removeProperties` function, you should create a copy of the `newObject` before deleting properties to avoid mutating the original object passed as an argument.